File: backend/app/schemas/class_schema.py
import uuid
from pydantic import BaseModel
from typing import Optional

class ClassBase(BaseModel):
    name: str
    section: str
    grade_level: int
    academic_year: str
    max_students: Optional[int] = 40

class ClassCreate(ClassBase):
    pass

class ClassUpdate(ClassBase):
    pass

class Class(ClassBase):
    id: uuid.UUID
    school_id: uuid.UUID
    class_teacher_id: Optional[uuid.UUID] = None

    class Config:
        from_attributes = True

File: frontend/school-portal/src/schemas/class_schema.ts

import { z } from 'zod';

export const ClassSchema = z.object({
  id: z.string(),
  name: z.string(),
  section: z.string(),
  grade_level: z.number(),
  school_id: z.string(),
  academic_year: z.string(),
  class_teacher_id: z.string(),
  max_students: z.number(),
});

export const ClassCreateSchema = ClassSchema.omit({ id: true });
export const ClassUpdateSchema = ClassSchema.partial();

export type Class = z.infer<typeof ClassSchema>;
export type ClassCreate = z.infer<typeof ClassCreateSchema>;
export type ClassUpdate = z.infer<typeof ClassUpdateSchema>;
